pkg/util/ec2/ec2.go

```diff
@@ -48,6 +48,29 @@ func getMetadataItem(endpoint string) (string, error) {
 	return string(all), nil
 }
 
+// GetClusterName returns the name of the cluster containing the current EC2 instance
+func GetClusterName() (string, error) {
+	tags, err := GetTags()
+	if err != nil {
+		return "", fmt.Errorf("unable to retrieve clustername from EC2: %s", err)
+	}
+
+	var clusterName string
+	for _, tag := range tags {
+		if strings.HasPrefix(tag, "kubernetes") { // tag key format: kubernetes.io/cluster/clustername"
```

Comment:
What happens if I set a node tag `kubernetes.io/role/master:1` in my cloudformation template like some deployment scripts seem to do?

Reply:
good point, i'd change it to `strings.HasPrefix(tag, "kubernetes.io/cluster/")`